admin-events/src/test-utils.js

```diff
@@ -0,0 +1,220 @@
+  const contractObjects = {};
+  abi.forEach((entry) => {
+    if (entry.type === objectType) {
+      contractObjects[entry.name] = entry;
+    }
+  });
+  return contractObjects;
+}
+
+function getEventFromConfig(abi, events) {
+  let eventInConfig;
+  let eventNotInConfig;
+  let findingType;
+  let findingSeverity;
+
+  const eventsInConfig = Object.keys(events);
+  const eventObjects = getObjectsFromAbi(abi, 'event');
+  Object.keys(eventObjects).forEach((name) => {
+    if ((eventNotInConfig !== undefined) && (eventInConfig !== undefined)) {
+      return;
+    }
+
+    if ((eventsInConfig.indexOf(name) === -1) && (eventNotInConfig === undefined)) {
+      eventNotInConfig = eventObjects[name];
+    }
+
+    if ((eventsInConfig.indexOf(name) !== -1) && (eventInConfig === undefined)) {
+      eventInConfig = eventObjects[name];
+      findingType = events[name].type;
+      findingSeverity = events[name].severity;
+    }
+  });
+  return {
+    eventInConfig, eventNotInConfig, findingType, findingSeverity,
+  };
+}
+
+function getExpressionOperand(operator, value, expectedResult) {
+  // given a value, an operator, and a corresponding expected result, return a value that
+  // meets the expected result
+  let leftOperand;
+  /* eslint-disable no-case-declarations */
+  if (BigNumber.isBigNumber(value)) {
+    switch (operator) {
+      case '>=':
+        if (expectedResult) {
+          leftOperand = value.toString();
+        } else {
+          leftOperand = value.minus(1).toString();
+        }
+        break;
+      case '<=':
+        if (expectedResult) {
+          leftOperand = value.toString();
+        } else {
+          leftOperand = value.plus(1).toString();
+        }
+        break;
+      case '===':
+        if (expectedResult) {
+          leftOperand = value.toString();
+        } else {
+          leftOperand = value.minus(1).toString();
+        }
+        break;
+      case '>':
+      case '!==':
+        if (expectedResult) {
+          leftOperand = value.plus(1).toString();
+        } else {
+          leftOperand = value.toString();
+        }
+        break;
+      case '<':
+        if (expectedResult) {
+          leftOperand = value.minus(1).toString();
+        } else {
+          leftOperand = value.plus(1).toString();
+        }
+        break;
+      default:
+        throw new Error(`Unknown operator: ${operator}`);
+    }
+  } else if (ethers.utils.isHexString(value)) {
+    switch (operator) {
+      case '===':
+        if (expectedResult) {
+          leftOperand = value;
+        } else {
+          leftOperand = ethers.constants.AddressZero;
+        }
+        break;
+      case '!==':
+        if (expectedResult) {
+          leftOperand = ethers.constants.AddressZero;
+        } else {
+          leftOperand = value;
+        }
+        break;
+      default:
+        throw new Error(`Unsupported operator ${operator} for hexString comparison`);
+    }
```

Comment:
The method `ethers.utils.isHexString` will check that the argument is of type `string` and starts with `0x` and contains only characters `0-9A-Fa-f` afterward.  Because an address string meets these qualifications, the `else if` block after this will never be run.

It probably makes sense to completely deprecate the `isAddress` function in favor of the `ethers` method.  There is an optional second argument to `ethers.utils.isHexString` to specify the length of the hex string, allowing us to achieve the same functionality as `isAddress`.

Also, lines 108 and 113 will work for every case except where `value` is equal to `ethers.constants.AddressZero`.

Reply:
I ended up wrapping a `ether.utils.isHexString(..., 20)` call in the `isAddress` function